src/libraries/System.Net.Security/src/System/Net/Security/Pal.OSX/SafeDeleteSslContext.cs

```diff
@@ -153,17 +160,17 @@ protected override void Dispose(bool disposing)
 
         private unsafe int WriteToConnection(void* connection, byte* data, void** dataLength)
         {
-            ulong toWrite = (ulong)*dataLength;
-            byte* readFrom = data;
+            ulong length = (ulong)*dataLength;
+            Debug.Assert(length <= int.MaxValue);
+
+            int toWrite = (int)length;
+            var inputBuffer = new ReadOnlySpan<byte>(data, toWrite);
```

Comment:
If this preempts right before acquiring the lock below, and another thread disposes of this context, `_outputBuffer` will be in a disposed state. Is this OK?